tests/test_database.py

```diff
@@ -146,14 +146,37 @@ def test_results(db, result):
     db.wipe_results()
     assert list(db.get_results()) == []
 
+    # Insert a valid non-logging result
+    db.insert_results([result])
+    db.insert_results([result])
+    log_component = {
+        'component1': 'level_info',
+    }
+    result['meta']['log_component'] = log_component
+
+    # Insert a valid logging result
+    db.insert_results([result])
+
+    # This should return only the logging result
+    assert list(db.get_results(log_component=log_component)) == [result]
+
+    # This should also return the logging result as level_info is same as info|error|warn|debug
+    assert list(db.get_results(log_component={'component1':'info|error|warn|debug'})) == [result]
+
+    # This should return an empty list as there are no results that match the provided log_component
+    assert list(db.get_results(log_component={'component1':'logic'})) == []
```

Comment:
https://www.techiedelight.com/check-list-empty-python/
```suggestion
    assert not list(db.get_results(log_component={'component1':'logic'}))
```